ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -74,7 +90,9 @@ queryEncodeNodeToClient codecConfig queryVersion blockVersion (SomeSecond query)
     TopLevelQueryDisabled -> encodeBlockQuery
       (case query of
         BlockQuery blockQuery -> blockQuery
+        GetSystemStart -> error "TODO: handle not available in this version"
       )
+    TopLevelQueryV2 -> error "TODO: need to add the BlockQuery wrapper encoding"
```

Comment:
In this case we should use a tagged union serialization similar to https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs#L213

Since the tag only appears when the query version is `/= TopLevelQueryDisabled`, we stay backwards compatible. Does that make sense, @dcoutts?